src/iso_alloc_util.c

```diff
@@ -48,16 +55,39 @@ INTERNAL_HIDDEN void *create_guard_page(void *p) {
         }
     }
 
-    /* Use g_page_size here because we could be
-     * calling this while we setup the root */
     mprotect_pages(p, g_page_size, PROT_NONE);
-    madvise(p, g_page_size, MADV_DONTNEED);
     name_mapping(p, g_page_size, GUARD_PAGE_NAME);
     return p;
 }
 
+/* Assumes p is page aligned and surrounded by guard pages */
+INTERNAL_HIDDEN void unmap_guarded_pages(void *p, size_t size) {
+    munmap(p - g_page_size, size + (g_page_size << 1));
+}
+
+/* Assumes size for guard pages is NOT accounted for.
+ * Returns a pointer to first RW page */
```

Comment:
Maybe add a the fact that the returned page is sandwitched between two guard ones?